wholesym/src/helper.rs

```diff
@@ -897,6 +907,32 @@ impl FileAndPathHelper for Helper {
 
         Ok(paths)
     }
+
+    fn get_symbol_map_for_library(
+        &self,
+        info: &LibraryInfo,
+    ) -> Option<(Self::FL, Box<dyn SymbolMapTrait + Send + Sync>)> {
+        if let Some(precog_helper) = self.config.precog_helper.as_ref() {
```

Comment:
I would just have a `HashMap<(String, DebugId), Arc<dyn SymbolMapTrait + Send + Sync>>` stored inside the helper. And then you would add those with `wholesym::SymbolManager::add_precog_symbol_map` similarly to how we add known libs. This wouldn't require an extra trait.